### PR TITLE
Feat/glif 39 refactor multisig navigation

### PR DESCRIPTION
**Pull request by Schwartz10**

### CODE REVIEW

File: components/Msig/AddRmSigners/Base.jsx [hunk 1]
```diff
@@ -356,13 +361,15 @@ const ManipulateSignersHOC = method => {
...
+    onClose: PropTypes.func.isRequired,
+    onComplete: PropTypes.func.isRequired,
     signers: PropTypes.arrayOf(
       PropTypes.shape({
         account: ADDRESS_PROPTYPE,
         id: ADDRESS_PROPTYPE
       })
-    )
+    ),
+    cid: ADDRESS_PROPTYPE
```
Comment (Schwartz10) [inline components/Msig/AddRmSigners/Base.jsx:367]: This should just be `PropTypes.string.isRequired`. A `cid` is not a filecoin address.

File: components/Msig/AddRmSigners/Base.jsx [hunk 2]
```diff
@@ -47,6 +47,11 @@ const ManipulateSignersHOC = method => {
     const [gasInfo, setGasInfo] = useState(emptyGasInfo)
     const [frozen, setFrozen] = useState(false)
 
+    // todo
+    // console.log(
+    //   `TODO: change UI auto select the id to remove. Selected: ${cid}`
+    // )
+
```
Comment (Schwartz10) [inline components/Msig/AddRmSigners/Base.jsx]: Let's make sure to get rid of this when you're ready

File: components/Msig/Home/AccountSummary.jsx [hunk 3]
```diff
@@ -15,137 +22,143 @@ import {
...
+            onClick={() => {
+              gotoRouteWithKeyUrlParams(router, PAGE_MSIG_CHANGE_OWNER)
+            }}
+            title='Change Signer'
+            height={6}
+            maxWidth={10}
+            minWidth={9}
+            borderRadius={6}
+          />
+        )}
```
Comment (Schwartz10) [inline components/Msig/Home/AccountSummary.jsx]: This doesn't go here anymore - the "edit" button alongside each Signer row is the new way to "change signer".

File: components/Msig/Home/Address.jsx [hunk 4]
```diff
@@ -44,20 +52,19 @@ const Address = ({ address, label, glyphAcronym }) => {
...
       <Box flexGrow='1'>
         <Label fontSize={1}>{label}</Label>
-        <CopyAddress
-          justifyContent='space-between'
-          color='core.nearblack'
-          address={address}
-        />
+        <CopyAddress color='core.nearblack' address={address} />
       </Box>
+      {onClose && <ButtonClose role='button' type='button' onClick={onClose} />}
```
Comment (Schwartz10) [inline components/Msig/Home/Address.jsx:56]: I understand why you're doing this, and I don't have any specific problem with approaching it this way, but it just _feels_ out of place to me for some reason. Not sure what to do yet, but can leave for now unless you think of something better.

File: components/Msig/Home/Balances.jsx [hunk 5]
```diff
@@ -1,10 +1,14 @@
...
 import styled from 'styled-components'
-import { func } from 'prop-types'
+import { useRouter } from 'next/router'
 import { Box, Button, Num, Title } from '../../Shared'
 import { FILECOIN_NUMBER_PROP } from '../../../customPropTypes'
 import makeFriendlyBalance from '../../../utils/makeFriendlyBalance'
 
+import { PAGE_MSIG_WITHDRAW } from '../../../constants'
+
+import { gotoRouteWithKeyUrlParams } from '../../../utils/urlParams'
```
Comment (Schwartz10) [inline components/Msig/Home/Balances.jsx]: Pedantic but lets just squeeze these together with no enters

File: components/Msig/Home/State.jsx [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
-import {
-  Box,
-  Button,
-  Title,
-  Text,
-  Menu,
-  MenuItem,
-  IconGlif
-} from '../../Shared'
+import { Box, Title, Menu, MenuItem, IconGlif } from '../../Shared'
```
Comment (Schwartz10) [inline components/Msig/Home/State.jsx]: Let's import things from `@glif/react-components` as much as we can!

File: components/Msig/Home/index.jsx [hunk 7]
```diff
@@ -33,50 +44,67 @@ const MsigHome = () => {
...
+              cid={router.query.cid || null}
+            />
+          )}
+        {!msig.loading &&
+          !messagesPending &&
+          childView === PAGE_MSIG_CHANGE_OWNER && (
+            <ChangeOwner
+              close={() => {
+                gotoRouteWithKeyUrlParams(router, PAGE_MSIG_CHANGE_OWNER)
+              }}
```
Comment (Schwartz10) [inline components/Msig/Home/index.jsx]: Marking this as a place we'll need to stub the same onComplete, onClose functionality..

File: components/Msig/Home/index.jsx [hunk 8]
```diff
@@ -1,27 +1,38 @@
-import React, { useState } from 'react'
+import React from 'react'
 import { useSelector } from 'react-redux'
+import { useRouter } from 'next/router'
 import { useMsig } from '../../../MsigProvider'
 import Withdraw from '../Withdraw'
 import ChangeOwner from '../ChangeOwner'
 import { AddSigner, RemoveSigner } from '../AddRmSigners'
 import { Box, LoadingScreen } from '../../Shared'
```
Comment (Schwartz10) [inline components/Msig/Home/index.jsx]: Let's try and import from the glif react components

**Review by Schwartz10 — COMMENTED** [review]
A couple high level things:

1. I think we should do for each of the multisig pages what we did for the send page - remember how you ripped the send page out of the home page, and just wrapped it in the `PageWrapper` component you pulled into the Glif react components? Let's do that for each of these multisig pages, so we don't need to use MsigHome to render everything.
2. This should make the active/inactive navigation tabs quite easy, here's what I'd do:
- Create a new component inside `/components/Msig` called `Nav`
- `Nav` should render the `NavLinks`.
- `Nav` is responsible for reading the router via `useRouter` and passing the active prop to the `NavLink`
- `Nav` gets rendered on each page, separately
3. Please uncommit any files that don't have to do with with this PR - like small linter changes in files that you wouldn't normally have touched. This is a big PR as it is, so we should keep it to the necessary changes. A major linting fix is coming this weekend.
4. In any file you touch, let's _try_ to import as many components as we can from the @glif/react-component library. A good strategy for doing this is:
- Run the tests for that file (they should pass)
- Change the import from 'shared' to 'react-components'
- Run the tests for that file
- If the tests pass, great. integration complete
- If the tests fail, lets see which component it fails on, and create a new Jira ticket to assess the diff of the shared directory in the wallet vs the react component library (we already know from earlier today that the Input component needs an assessment.
5. The Owners page can now include the `Confirmation Threshold` part of the designs - you can access this through `const { NumApprovalsThreshold } = useMsig()`

File: components/Msig/Home/Address.jsx (hunk 4 above)
Comment (erchaves) [reply components/Msig/Home/Address.jsx:56]: oh, well, `onClose` is definitely the wrong name I'm seeing. We could rename this to removeAddress or something. But do you mean that the UI is also wrong? We can remove this if you want.

File: components/Msig/Home/State.jsx (hunk 6 above)
Comment (erchaves) [reply components/Msig/Home/State.jsx]: nice, cool, ill switch all those

File: components/Msig/Home/Address.jsx (hunk 4 above)
Comment (Schwartz10) [reply components/Msig/Home/Address.jsx:56]: Yeah i like `onRemoveAddress` (let's stick with `on__` if we're going that way). I haven't seen the UI let, need to add a signer to my multisig

File: components/Msig/Home/State.jsx (hunk 6 above)
Comment (Schwartz10) [reply components/Msig/Home/State.jsx]: i would make sure to do it slowly just so you know if something broke along the way (i clearly didn't follow this advice when importing inputs into the app)

**Comment by Schwartz10** [comment]
Closing in favor of #855

**Pull request closed without merging**